zookeeper-server/src/main/java/org/apache/zookeeper/client/ZooKeeperSaslClient.java

```diff
@@ -77,7 +77,7 @@
      */
     @Deprecated
     public static boolean isEnabled() {
-        return Boolean.valueOf(System.getProperty(ZKClientConfig.ENABLE_CLIENT_SASL_KEY, ZKClientConfig.ENABLE_CLIENT_SASL_DEFAULT));
+        return Boolean.getBoolean(System.getProperty(ZKClientConfig.ENABLE_CLIENT_SASL_KEY, ZKClientConfig.ENABLE_CLIENT_SASL_DEFAULT));
```

Comment:
Reverted as `Boolean.parseBoolean`. It should never be `Boolean.getBoolean("true"|"false")` which would nearly always returns false.

Reply:
Should never do refactor then.